internal/infrastructure/datastore/db.go

```diff
@@ -48,6 +48,12 @@ func NewDB(cfg *config.Config) (db *gorm.DB) {
 		slog.Error("Error creating database connector", slog.Any("error", err))
 		return nil
 	}
+	sqlDb, err := db.DB()
+	if err != nil {
+		slog.Error("Error getting sql driver", slog.String("error", err.Error()))
+		return nil
+	}
+	sqlDb.SetMaxOpenConns(cfg.Database.MaxOpenConns)
```

Comment:
Also interesting would be to expose _current_ number of DB connections in the prometheus metrics (if we don't already do that).  But it can be done as a separate change.

Reply:
I agree that we want to print the value. But is it good to print it here? I.e. it would make sense to me to print the DB server hostname and port as well. 

From that PoV, it was the reason why I mention printing whole config without secrets after it is loaded - on Info level.  WDYT?

I'm also thinking to merge this and solve the print separately.

Reply:
That's fine with me.  Let's merge it and treat the logging/debug as a separate change.